game/menu/gamemanager.gd

```diff
@@ -13,22 +22,164 @@ func _ready():
+		# put contents from file in the cache
+		GAME_DATA_CACHE[cache_key] = parse_result.result
+
+	return GAME_DATA_CACHE[cache_key]
+
+
+## Save the changes to the Dictionary returned by get_game_data()
+## This method is automatically called when a game ends.
+func save_game_data():
+	assert(_last_loaded_game != null)  # this should only be called if _last_loaded_game is set
+	_save_data("game_data", _load_data("game_data", _last_loaded_game), _last_loaded_game)
+
+
+## Get the game data for the current player and the current game.
+## To save data. Just modify the returned Dictionary and call save_game_data().
+func get_game_data() -> Dictionary:
+	assert(_last_loaded_game != null)
+	return _load_data("game_data", _last_loaded_game)
+
+
+func get_current_player() -> String:
+	return "p"  # in future add here more logic
+
+
+## Get the time the current player played the currently running game last.
+func get_last_played(game_id = null):
+	game_id = _last_loaded_game if game_id == null else game_id
+	assert(game_id != null)
+	var data = _load_data("game_meta_data", game_id)
+	if not "last_played" in data:
+		return null
+	var dt = OS.get_datetime_from_unix_time(data["last_played"])
+	return (
+		"%04d-%02d-%02d %02d:%02d:%02d UTC"
+		% [dt["year"], dt["month"], dt["day"], dt["hour"], dt["minute"], dt["second"]]
+	)
+
+
+## Get time the current player played the currently running game.
+func get_played_time(game_id = null) -> float:
+	game_id = _last_loaded_game if game_id == null else game_id
+	assert(game_id != null)
+	var data = _load_data("game_meta_data", game_id)
+	if not "played_time" in data:
+		return 0.0
+	return data["played_time"]
+
+
+## Get the high_score of the current player for the currently running game.
+func get_high_score(game_id = null):
+	game_id = _last_loaded_game if game_id == null else game_id
+	if not game_id:  # game_id should be the folder_name, not null or ""
+		return null
+	var data = _load_data("game_meta_data", game_id)
+
+	if not "scores" in data:
+		return null
+
+	if data["scores"].size() == 0:
+		return null
+
+	return data["scores"][-1][0]
+
+
 # return to the level select
-func end_game(message := "", _status = null):
+func end_game(message := "", score = null, _status = null):
 	get_tree().change_scene("res://menu/emptySzene.tscn")
 	_main.show()
+
+	save_game_data()  # save the data stored while the game was playing
+
+	assert(_last_loaded_game != null)  # should always be set here
+
+	var data = _load_data("game_meta_data", _last_loaded_game)
+
+	data["last_played"] = OS.get_unix_time()
+
+	if not "played_time" in data:
+		data["played_time"] = 0
+	data["played_time"] += (OS.get_ticks_msec() - _started_playing_game) / 1000.0
+
+	if score != null:
+		if not "scores" in data:
+			data["scores"] = []
+		data["scores"].append([score, data["last_played"]])
+		data["scores"].sort()
+
+	_save_data("game_meta_data", data, _last_loaded_game)
+
+	GAME_DISPLAYS[_last_loaded_game].update_text()
```

Comment:
Might be a way to add multiple profiles (there was an PR right?) and keep the names here as well.

Reply:
I added a method that returns the current player (currently not changing), that can later be changed for multiple profiles